ukvm/ukvm_elf.c

```diff
@@ -196,7 +196,11 @@ void ukvm_elf_load(const char *file, uint8_t *mem, size_t mem_size,
         prot = PROT_NONE;
         if (phdr[ph_i].p_flags & PF_R)
             prot |= PROT_READ;
+/* GDB clients can change memory, and software breakpoints work by replacing
+ * instructions with int3's. */
+#ifndef UKVM_MODULE_GDB
         if (phdr[ph_i].p_flags & PF_W)
+#endif
```

Comment:
There's a better way to do this. Rather than cluttering the code here, drop this change and add the following to `setup()` in the GDB module just before the call to `gdb_stub_start()`:
````
    if (mprotect(hv->mem, hv->mem_size, PROT_READ | PROT_WRITE | PROT_EXEC) == -1)
        err(1, "GDB: Cannot remove guest memory protection");
````
For reference, calling `mprotect()` more than once on the same or enclosing region is explicitly allowed, at least on Linux, precisely for this purpose.